source/Cosmos.HAL2/Global.cs

```diff
@@ -45,16 +45,45 @@ static public void Init(TextScreenBase textScreen)
       ACPI.Start();
 
       mDebugger.Send("Done initializing Cosmos.HAL.Global");
-
-      mDebugger.Send("ATA Primary Master");
-      InitAta(Ata.ControllerIdEnum.Primary, Ata.BusPositionEnum.Master);
-
-      //TODO Need to change code to detect if ATA controllers are present or not. How to do this? via PCI enum?
-      // They do show up in PCI space as well as the fixed space.
-      // Or is it always here, and was our compiler stack corruption issue?
-      mDebugger.Send("ATA Secondary Master");
-      InitAta(Ata.ControllerIdEnum.Secondary, Ata.BusPositionEnum.Master);
-      //InitAta(BlockDevice.Ata.ControllerIdEnum.Secondary, BlockDevice.Ata.BusPositionEnum.Slave);
+      
+      // Class 0x01 = Mass Storage Controllers' Class code
+      // Subclass 0x00 = Subclass code of SCSI Controller
+      // Subclass 0x01 = Subclass code of IDE Controller
+      // Subclass 0x04 = Subclass code of RAID Controller
+      // Subclass 0x06 = Subclass code of AHCI Controller
+      if (PCI.GetDeviceClass(0x01, 0x00) != null)
+      {
+        mDebugger.Send("SCSI isn't supported yet");
+        Console.WriteLine("SCSI Controller not supported yet");
+        Console.WriteLine("Booting without ATA Initialization");
+        Console.WriteLine("FAT cannot be used while ATA isn't initialized"); 
+      }
+      else if (PCI.GetDeviceClass(0x01, 0x01) != null)
```

Comment:
I think that this should be the first check. If ATA is detected, we don't really care about the others.